langkit/templates/c_api/pkg_main_spec_ada.mako

```diff
@@ -520,6 +521,16 @@ private package ${ada_lib_name}.Implementation.C is
       % endif
    % endfor
 
+   --------------------
+   -- Iterator types --
+   --------------------
+
```

Comment:
I suggest to insert here documentation that will be common to all iterator types in this section: describe what iterators are, that these types have ref-counting reference semantics, and of course describe safety aspects once they are implemented.

Reply:
… also document the same way the Next functions